app/jobs/prepare_move_notifications_job.rb

```diff
@@ -38,9 +39,15 @@ def build_notification_id(subscription, type_id, topic, action_name)
           ).id }
   end
 
-  def should_email?(move, action_name)
-    # NB: only notify by email on move requested and move cancellation
-    [Move::MOVE_STATUS_REQUESTED, Move::MOVE_STATUS_CANCELLED].include?(move.status) && %w(create update_status).include?(action_name)
+  def should_email?(move)
+    # NB: only email for current moves (not back-dated ones):
+    #   * move.status must be Requested or Cancelled (not Proposed), AND
+    #   * move.date is not in the past OR move.to_date is not in the past
+
+    [Move::MOVE_STATUS_REQUESTED, Move::MOVE_STATUS_CANCELLED].include?(move.status) &&
```

Comment:
Would it be cleaner to move this to question to the Move model?

```ruby
move.current?
```

Reply:
It would be easier to unit test

Reply:
yes, good point 👍